backend/controllers/Event.js

```diff
@@ -65,7 +85,96 @@ const createEvent = async (req, res) => {
  * req: request object containing user id and event id.
  * res: response object to return status 200.
  */
+/**
+ * NOTES:
+ * - might implement time caped. Rigth now, expected that start and end time is in the same day
+ * - anyone can click finish for now
+ */
+
+const finishEvent = async (req, res) => {
+  try {
+    const eventId = req.params.id;
+
+    // the point will only be distributed if user finish the event before the end time
+
+    const event = await Event.findById(eventId);
+
+    // if event is completed, wont do anything
+    if (event.status === 'completed') {
+      res.status(400).json({ message: 'Event already completed' });
+    }
```

Comment:
NIT: if you want to learn better practice, look into enum. Change your event model, status field from string to enum of specific values like completed, in progress, not started, etc

Reply:
Yes enum is already in model anh :3